pkg/config/config_test.go

```diff
@@ -38,32 +39,112 @@ func TestLoad(t *testing.T) {
 
 // Get auth properties from env vars
 func TestGetAuthFromEnvVar(t *testing.T) {
-	var syncConfig api.Config
-	cfgFile := "../../testdata/example-config-no-auth.yaml"
-	viper.SetConfigFile(cfgFile)
-	os.Setenv("SOURCE_AUTH_USERNAME", "sUsername")
-	os.Setenv("SOURCE_AUTH_PASSWORD", "sPassword")
-	os.Setenv("TARGET_AUTH_USERNAME", "tUsername")
-	os.Setenv("TARGET_AUTH_PASSWORD", "tPassword")
-	// If a config file is found, read it in.
-	if err := viper.ReadInConfig(); err != nil {
-		t.Fatalf("error reading config file: %+v", err)
-	}
-	if err := Load(&syncConfig); err != nil {
-		t.Fatalf("error loading config file")
+	tests := map[string]struct {
+		inputFile string
+		envVars   map[string]string
+		expected  map[string]string
```

Comment:
You should compare the final output, not part of it separately. Otherwise, you may introduce bugs in other fields (if you extended the object, for example) and don't notice a side effect. In this case, the `api.Auth` object. This way you need to compare with `reflect` or `proto` packages:

```
expectedSourceAuth *api.Auth
expectedTargetAuth *api.Auth

...

			"example-config-no-auth.yaml",
			map[string]string{
				"SOURCE_AUTH_USERNAME": "sUsername",
				"SOURCE_AUTH_PASSWORD": "sPassword",
				"TARGET_AUTH_USERNAME": "tUsername",
				"TARGET_AUTH_PASSWORD": "tPassword",
			},
			&api.Auth{Username: "sUsername", Password: "sPassword"},
			&api.Auth{Username: "tUsername", Password: "tPassword"},

...

			if got, want := source.Repo.Auth, tc.expectedSourceAuth; !proto.Equal(got, want) {
				t.Errorf("got: %+v; want %+v", got, want)
			}
```

Note that the pattern in Go is to use `got` and `want` in tests.

Reply:
done!